File: src/theme.ts
// Theme abstraction for simple-diff
// Currently using Catppuccin Mocha palette
// Replace this file to change the color scheme

export interface Theme {
	// Base colors
	base: string;
	mantle: string;
	crust: string;
	surface0: string;
	surface1: string;
	surface2: string;

	// Text colors
	text: string;
	subtext0: string;
	subtext1: string;
	overlay0: string;
	overlay1: string;
	overlay2: string;

	// Accent colors
	blue: string;
	lavender: string;
	sapphire: string;
	sky: string;
	teal: string;
	green: string;
	yellow: string;
	peach: string;
	maroon: string;
	red: string;
	mauve: string;
	pink: string;
	flamingo: string;
	rosewater: string;

	// Semantic colors (derived from accent colors)
	primary: string;
	secondary: string;
	success: string;
	warning: string;
	error: string;

	// Diff-specific colors
	diffAdded: string;
	diffAddedBg: string;
	diffRemoved: string;
	diffRemovedBg: string;
	diffContext: string;
	diffContextBg: string;

	// UI element colors
	border: string;
	borderFocused: string;
	inputBg: string;
	inputFocusedBg: string;
	headerBg: string;
	footerBg: string;
}

// Catppuccin Mocha palette
export const catppuccinMocha: Theme = {
	// Base colors
	base: "#1e1e2e",
	mantle: "#181825",
	crust: "#11111b",
	surface0: "#313244",
	surface1: "#45475a",
	surface2: "#585b70",

	// Text colors
	text: "#cdd6f4",
	subtext0: "#a6adc8",
	subtext1: "#bac2de",
	overlay0: "#6c7086",
	overlay1: "#7f849c",
	overlay2: "#9399b2",

	// Accent colors
	blue: "#89b4fa",
	lavender: "#b4befe",
	sapphire: "#74c7ec",
	sky: "#89dceb",
	teal: "#94e2d5",
	green: "#a6e3a1",
	yellow: "#f9e2af",
	peach: "#fab387",
	maroon: "#eba0ac",
	red: "#f38ba8",
	mauve: "#cba6f7",
	pink: "#f5c2e7",
	flamingo: "#f2cdcd",
	rosewater: "#f5e0dc",

	// Semantic colors
	primary: "#89b4fa",
	secondary: "#cba6f7",
	success: "#a6e3a1",
	warning: "#f9e2af",
	error: "#f38ba8",

	// Diff-specific colors
	diffAdded: "#a6e3a1",
	diffAddedBg: "#1e3a2f",
	diffRemoved: "#f38ba8",
	diffRemovedBg: "#3e2a35",
	diffContext: "#cdd6f4",
	diffContextBg: "#1e1e2e",

	// UI element colors
	border: "#45475a",
	borderFocused: "#89b4fa",
	inputBg: "#313244",
	inputFocusedBg: "#45475a",
	headerBg: "#181825",
	footerBg: "#181825",
};

// Export the active theme
export const theme: Theme = catppuccinMocha;

File: rust/src/theme.rs
// Theme abstraction for difftui
// Currently using Catppuccin Mocha palette

use ratatui::style::Color;

#[allow(dead_code)]
pub struct Theme {
    // Base colors
    pub base: Color,
    pub mantle: Color,
    pub crust: Color,
    pub surface0: Color,
    pub surface1: Color,
    pub surface2: Color,

    // Text colors
    pub text: Color,
    pub subtext0: Color,
    pub subtext1: Color,
    pub overlay0: Color,
    pub overlay1: Color,
    pub overlay2: Color,

    // Accent colors
    pub blue: Color,
    pub lavender: Color,
    pub sapphire: Color,
    pub sky: Color,
    pub teal: Color,
    pub green: Color,
    pub yellow: Color,
    pub peach: Color,
    pub maroon: Color,
    pub red: Color,
    pub mauve: Color,
    pub pink: Color,
    pub flamingo: Color,
    pub rosewater: Color,

    // Semantic colors
    pub primary: Color,
    pub secondary: Color,
    pub success: Color,
    pub warning: Color,
    pub error: Color,

    // Diff-specific colors
    pub diff_added: Color,
    pub diff_added_bg: Color,
    pub diff_removed: Color,
    pub diff_removed_bg: Color,
    pub diff_context: Color,
    pub diff_context_bg: Color,

    // UI element colors
    pub border: Color,
    pub border_focused: Color,
}

// Catppuccin Mocha palette
pub const THEME: Theme = Theme {
    // Base colors
    base: Color::Rgb(27, 27, 41),
    mantle: Color::Rgb(24, 24, 37),
    crust: Color::Rgb(17, 17, 27),
    surface0: Color::Rgb(49, 50, 68),
    surface1: Color::Rgb(69, 71, 90),
    surface2: Color::Rgb(88, 91, 112),

    // Text colors
    text: Color::Rgb(205, 214, 244),
    subtext0: Color::Rgb(166, 173, 200),
    subtext1: Color::Rgb(186, 194, 222),
    overlay0: Color::Rgb(108, 112, 134),
    overlay1: Color::Rgb(127, 132, 156),
    overlay2: Color::Rgb(147, 153, 178),

    // Accent colors
    blue: Color::Rgb(137, 180, 250),
    lavender: Color::Rgb(180, 190, 254),
    sapphire: Color::Rgb(116, 199, 236),
    sky: Color::Rgb(137, 220, 235),
    teal: Color::Rgb(148, 226, 213),
    green: Color::Rgb(166, 227, 161),
    yellow: Color::Rgb(249, 226, 175),
    peach: Color::Rgb(250, 179, 135),
    maroon: Color::Rgb(235, 160, 172),
    red: Color::Rgb(243, 139, 168),
    mauve: Color::Rgb(203, 166, 247),
    pink: Color::Rgb(245, 194, 231),
    flamingo: Color::Rgb(242, 205, 205),
    rosewater: Color::Rgb(245, 224, 220),

    // Semantic colors
    primary: Color::Rgb(137, 180, 250),
    secondary: Color::Rgb(203, 166, 247),
    success: Color::Rgb(166, 227, 161),
    warning: Color::Rgb(249, 226, 175),
    error: Color::Rgb(243, 139, 168),

    // Diff-specific colors
    diff_added: Color::Rgb(166, 227, 161),
    diff_added_bg: Color::Rgb(30, 58, 47),
    diff_removed: Color::Rgb(243, 139, 168),
    diff_removed_bg: Color::Rgb(62, 42, 53),
    diff_context: Color::Rgb(205, 214, 244),
    diff_context_bg: Color::Rgb(30, 30, 46),

    // UI element colors
    border: Color::Rgb(69, 71, 90),
    border_focused: Color::Rgb(137, 180, 250),
};
